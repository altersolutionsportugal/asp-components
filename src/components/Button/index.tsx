// eslint-disable-next-line import/no-unresolved
import { ButtonProps } from '../../@types/Button'

const Button = ({ children, onClick }: ButtonProps) => (
  <div aria-hidden="true" onClick={onClick}>
    { children }
  </div>
)

export default Button
