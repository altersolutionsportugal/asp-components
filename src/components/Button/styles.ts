import { styled } from '../../styles/stitches.config'

export const ButtonContainer = styled('button', {
  backgroundColor: '$gray400',
  borderRadius: '9999px',
  fontSize: '13px',
  padding: '10px 15px',
  '&:hover': {
    backgroundColor: 'lightgray',
  },
})
