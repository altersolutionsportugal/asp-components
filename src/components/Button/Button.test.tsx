import { render } from '@testing-library/react'

import Button from '.'

describe('renders component Button', () => {
  it('should create snapShot', () => {
    const { container } = render(<Button><h1>Hello</h1></Button>)

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render component with children', () => {
    const { getByText } = render(<Button><h1>Hello</h1></Button>)

    expect(getByText('Hello')).toBeInTheDocument()
  })
})
