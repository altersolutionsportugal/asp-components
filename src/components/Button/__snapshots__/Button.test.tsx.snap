// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component Button should create snapShot 1`] = `
<button
  aria-hidden="true"
  class="c-cGLqUE"
>
  <h1>
    Hello
  </h1>
</button>
`;
